### PR TITLE
Don't change rootElement when received node is the same

### PR DESCRIPTION
**Pull request by goncalo-matos**
Issue:
storybook-html's render is always rewriting the root element's content even when it's not a different component. This appeared to be a problem for dynamic class changes (as in changing a prop in a AngularJS component) as it is removing the dom node and adding it again which kills stuff like animations (since from the browser's perspective this is not a change in the class attribute and it is a new node).

Props to @svipatov for finding the issue and help developing this PR.

## What I did
Checked the reference of the received DOM node, if it is the same then the node was already mutated and there is no need to rewrite the root element.

## How to test

- Is this testable with Jest or Chromatic screenshots?
It could be, but I did not find any test for the changed file (I can add it if it's wanted)
- Does this need a new example in the kitchen sink apps?
No
- Does this need an update to the documentation?
No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by igor-dv** [comment]
Can you maybe share a usecase ? I can think of knobs changes, but for this, we have a `forceRender` parameter that can be helpful. [Here](https://github.com/storybooks/storybook/blob/next/app/react/src/client/preview/render.js#L53) is the example in `@storybook/react`

**Comment by igor-dv** [comment]
Maybe `forceRender` is a bit misleading name, but it means that the story was rerendered forcibly (and not navigated). In other words, when knobs are changed, this parameter will be `true`.

**Comment by Atekon** [comment]
Yeah, from your explanation it seems `forceRender` will do the trick. We'll change it and update/reopen the PR

File: app/html/src/client/preview/render.js [hunk 1]
```diff
@@ -3,13 +3,24 @@ import { stripIndents } from 'common-tags';
...
+  showError,
+  forceRender,
+}) {
   const component = story();
 
   showMain();
   if (typeof component === 'string') {
     rootElement.innerHTML = component;
   } else if (component instanceof Node) {
+    if (!forceRender) {
```
[the remark this answers is not in the document]
Comment (Atekon) [reply app/html/src/client/preview/render.js]: Yeah, you're right! sorry, was distracted

**Review by igor-dv — APPROVED** [review]

**Pull request merged**